pkg/service/events.go

```diff
@@ -0,0 +1,100 @@
+package service
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/keikoproj/lifecycle-manager/pkg/log"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type EventReason string
+type EventLevel string
+
+var (
+	// EventName is the default name for service events
+	EventName = "lifecycle-manager.%v"
+	// EventNamespace is the default namespace in which events will be published in
+	EventNamespace = "default"
+
+	EventLevelNormal  = "Normal"
+	EventLevelWarning = "Warning"
+
+	EventReasonLifecycleHookReceived  EventReason = "EventLifecycleHookReceived"
+	EventMessageLifecycleHookReceived             = "lifecycle hook for event %v was received, instance %v will begin processing"
+
+	EventReasonLifecycleHookProcessed  EventReason = "EventLifecycleHookProcessed"
+	EventMessageLifecycleHookProcessed             = "lifecycle hook for event %v has completed processing, instance %v gracefully terminated"
+
+	EventReasonLifecycleHookFailed  EventReason = "EventLifecycleHookFailed"
+	EventMessageLifecycleHookFailed             = "lifecycle hook for event %v has failed processing: %v"
+
+	EventReasonNodeDrainSucceeded  EventReason = "EventReasonNodeDrainSucceeded"
+	EventMessageNodeDrainSucceeded             = "node %v has been drained successfully as a response to a termination event"
+
+	EventReasonNodeDrainFailed  EventReason = "EventReasonNodeDrainFailed"
+	EventMessageNodeDrainFailed             = "node %v draining has failed: %v"
+
+	EventReasonTargetDeregisterSucceeded  EventReason = "EventReasonTargetDeregisterSucceeded"
+	EventMessageTargetDeregisterSucceeded             = "target %v:%v has successfully deregistered from target group %v"
+
+	EventReasonTargetDeregisterFailed  EventReason = "EventReasonTargetDeregisterFailed"
+	EventMessageTargetDeregisterFailed             = "target %v:%v has failed to deregistered from target group %v: %v"
+
+	EventReasonInstanceDeregisterSucceeded  EventReason = "EventReasonInstanceDeregisterSucceeded"
+	EventMessageInstanceDeregisterSucceeded             = "instance %v has successfully deregistered from classic-elb %v"
+
+	EventReasonInstanceDeregisterFailed  EventReason = "EventReasonInstanceDeregisterFailed"
+	EventMessageInstanceDeregisterFailed             = "instance %v has failed to deregistered from classic-elb %v: %v"
```

Comment:
Are these `EventReason`'s good candidates for constants?

Reply:
Yes, I don't see anything or any test that would need to change these.